www/js/diary/services.js

```diff
@@ -1005,22 +1010,17 @@ angular.module('emission.main.diary.services', ['emission.plugin.logger',
                          +"unprocessedTripList.length = "+unprocessedTripList.length);
               Array.prototype.push.apply(tripList, unprocessedTripList);
               console.log("After merge, returning trip list of size "+tripList.length);
-              return tripList;
+              return [tripList, modes, purposes];
             });
         } else {
-            return tripList;
+            return [tripList, modes, purposes];
         }
-      }).then(function(combinedTripList) {
-          var tq = { key: 'write_ts', startTs: 0, endTs: moment().endOf('day').unix(), };
-          return Promise.all([
-            UnifiedDataLoader.getUnifiedMessagesForInterval('manual/mode_confirm', tq),
-            UnifiedDataLoader.getUnifiedMessagesForInterval('manual/purpose_confirm', tq),
-            EnketoSurvey.getAllSurveyAnswers("manual/confirm_survey", { populateLabels: true }),
-          ]).then(function(results) {
+      }).then(function([combinedTripList, modes, purposes]) {
+          return EnketoSurvey.getAllSurveyAnswers("manual/confirm_survey", { populateLabels: true }).then(function(surveyAnswers) {
```

Comment:
You need to move this into the first call too. `getAllSurveyAnswers` calls `UnifiedDataLoader.getUnifiedMessagesForInterval` which also makes a remote call to the server. All remote calls to the server need to run in parallel in the first call to avoid the latency caused by the second call.